test/scale_optional_test.cpp

```diff
@@ -9,60 +9,54 @@
 #include <scale/scale.hpp>
 
 using scale::ByteArray;
-using scale::decode;
 using scale::DecodeError;
-using scale::encode;
 using scale::EncodeError;
-using scale::ScaleDecoderStream;
-using scale::ScaleEncoderStream;
+using scale::impl::memory::decode;
+using scale::impl::memory::encode;
+using Decoder = scale::backend::FromBytes;
 
-// TODO(Harrm): PRE-119 refactor to parameterized tests
 /**
  * @given variety of optional values
  * @when encodeOptional function is applied
  * @then expected result obtained
  */
-TEST(Scale, EncodeOptional) {
+TEST(Optional, Encode) {
   // most simple case
   {
-    ScaleEncoderStream s;
-    ASSERT_NO_THROW((s << std::optional<uint8_t>{std::nullopt}));
-    ASSERT_EQ(s.to_vector(), (ByteArray{0}));
+    ASSERT_OUTCOME_SUCCESS(encoded,
+                           encode(std::optional<uint8_t>{std::nullopt}));
+    ASSERT_EQ(encoded, (ByteArray{0}));
   }
 
   // encode existing uint8_t
   {
-    ScaleEncoderStream s;
-    ASSERT_NO_THROW((s << std::optional<uint8_t>{1}));
-    ASSERT_EQ(s.to_vector(), (ByteArray{1, 1}));
+    ASSERT_OUTCOME_SUCCESS(encoded, encode(std::optional<uint8_t>{1}));
+    ASSERT_EQ(encoded, (ByteArray{1, 1}));
   }
 
   // encode negative int8_t
   {
-    ScaleEncoderStream s;
-    ASSERT_NO_THROW((s << std::optional<int8_t>{-1}));
-    ASSERT_EQ(s.to_vector(), (ByteArray{1, 255}));
+    ASSERT_OUTCOME_SUCCESS(encoded, encode(std::optional<uint8_t>{-1}));
```

Comment:
Comment suggests that we encode negative int8_t, but in fact we encode uint8_t. Was that intentional?

Reply:
Hm... Looks like legacy code, I just have it changed for new way. I'll research ability to encode signed fixed-wight integers, may be it possible. For compact-integers it is definitely prohibited.

Reply:
Signed sized-width integers are supported